snakefile

```diff
@@ -1,18 +1,54 @@
 # Tick diet sequence assemblies: Overall workflow
-import os
-from re import search
+import csv
+import warnings
 
 ## SETUP:
-configfile: "config.yml"
+#configfile: "config.yml"
 
-# Find all samples to process:
-RAW_READS = glob_wildcards("data/reads/{run_id}/{filestem}_{samplenum}_{lane}_{read}_001.fastq.gz")
+REF_GENOM = "data/ref_genome/ref.fa"
 
-## Parse names:
-include: "makefiles/setup.smk"
+## Samples to process
+# Find all sample files in data directory:
+AVAILABLE_FILES = glob_wildcards("data/input/multi_refs/{run_id}/{sample}.bam")
+
+# Check qPCR results
+known_positives = []
+
+with open("data/metadata/SampleMetadata.csv", mode="r", encoding="utf-8-sig") as handle:
+    sample_file = csv.DictReader(handle)
+    
+    for line in sample_file:
+        if line["pcr result"].lower() == "pos":
+            known_positives.append(line["sample"])
+
+# <add code to check false positives here>

```

Comment:
If needed, could move the logic from `find_borrelia_positive_samples.sh` here too - just update `known_positives` to exclude further samples (but remember to check across all run_ids in case the sequencing just failed and was re-sequenced).

Reply:
do you mean to not include the bash script and instead do it in snakemake?

Reply:
how would snakemake use the known_positives variable downstream if snakemake uses wildcards to locate files in folders?

Reply:
After looking at the code more closely I think what you're suggesting is to map all files but only use the positives for the variant calling?

Reply:
No, you can use the existing alignments for this test as before, I just mean that since this bit of code builds up a list of samples to process (for both mapping and variant calling), one could modify that list of sample names directly rather than move the files. But it's just a suggestion, whichever implementation is easiest for you is fine.

Reply:
Remember that since the `combine_gvcfs` rule only combines files for positive samples, the upstream bam and vcf files for negative samples will never be created, simply because they are not needed/used anywhere in the pipeline. The wildcards being matched in `AVAILABLE_FILES` simply show where files are, but snakemake won't do anything with those files unless needed to create files requested in the terminal node (the `all` rule).

Reply:
OK, now I understand. So even if we don't specifically exclude the negative files in the mapping phase they still won't be made because they don't end up in the terminal node. Snakemake is interesting, because you would have to code this specifically in any other implementation hence my new directory idea.

Reply:
I like this implementation, but another reason I wanted to do this with a bash script and move the files was because checking for false positives is easier this way. To do it here, do you think it's possible to use the os module to call samtools inside the python loop? Or would we have to, for example, activate the conda environment inside the os command and immediately afterwards run the samtools check for false positives?

Reply:
I think I'll look into using pysam for it

Reply:
Since snakemake will be running inside the conda environment, custom python code should find samtools just like the regular snakemake rules do. But pysam would work too - just add the module to the conda environment, since it includes the python installation used by snakemake.